common/scripts/babel-conditional-preprocess.js

```diff
@@ -63,62 +74,96 @@ exports.default = babelHelper.declare((_api, opts) => {
       // As this only applies to TypeScript types, it is safe from a code-flow perspective: This does not enable any new
       // conditional business logic flows.
       TSType(path) {
-        Handle(path, annotations);
+        Handle(path, match, featureSet, stabilizedFeatureSet);
       },
 
       TSDeclareMethod(path) {
-        Handle(path, annotations);
+        Handle(path, match, featureSet, stabilizedFeatureSet);
       },
 
       Expression(path) {
-        Handle(path, annotations);
+        Handle(path, match, featureSet, stabilizedFeatureSet);
       },
 
       ClassMethod(path) {
-        Handle(path, annotations);
+        Handle(path, match, featureSet, stabilizedFeatureSet);
       },
 
       ClassProperty(path) {
-        Handle(path, annotations);
+        Handle(path, match, featureSet, stabilizedFeatureSet);
       },
     }
   };
 });
 
-function Handle(path, annotations) {
+function Handle(path, match, featureSet, stabilizedFeatureSet) {
   let { node } = path;
-  let removed = false;
-
-  for (const annotation of annotations) {
-    const {
-      match
-    } = annotation;
-
-    if (node.leadingComments && node.leadingComments.length > 0) {
-      for (const comment of node.leadingComments) {
-        if (comment.value.includes(match)) {
-          removed = true;
-        }
-        if (removed) {
-          comment.ignore = true;
-          // Comment will be inherit to next line even it is set to 'ignore'
-          // this will cause some unexpected removals of code
-          // clear the comment to ensure nothing gets wrong
-          comment.value = '';
-        }
-      }
-    }
 
-    if (removed) {
```

Comment:
The function scoped variable 'removed' should stay - it is intentional to remove all the following comment after the conditional compilation match

Imagine a case like this:

// conditional-compile-remove(stable)
/* This is the comment for the beta feature */
const betaFeauture = () => {}

The second comment should be also removed with the statement itself, without the function scoped the comment will still stay, and mess up our jsdoc for stable api doc

Reply:
This check is still there. It's just an early return (line 107 after the PR).

Reply:
Oh wait, now I understand the issue.